posthog/temporal/ai/session_summary/activities/a6_store_video_session_summary.py

```
@@ -0,0 +1,496 @@
+                "event_type": event_type,
+                "event_index": event_idx,
+                "session_id": session_id,
+                "event_uuid": event_uuid,
+            }
+        else:
+            # Fallback to synthetic event
+            key_action_event = {
+                "description": segment.description,
+                "abandonment": segment.abandonment_detected,
+                "confusion": segment.confusion_detected,
+                "exception": None,
+                "event_id": f"vid_{idx:04d}",
+                "timestamp": segment.start_time,
+                "milliseconds_since_start": start_ms,
+                "current_url": None,
+                "window_id": None,
+                "event": "video_segment",
+                "event_type": "video_analysis",
+                "event_index": idx,
+                "session_id": session_id,
+                "event_uuid": None,
+            }
+
+        key_actions.append(
+            {
+                "segment_index": idx,
+                "events": [key_action_event],
+            }
+        )
+
+    # Use LLM-provided segment outcomes, or fall back to generating from segment data
+    if analysis.segment_outcomes:
+        segment_outcomes = analysis.segment_outcomes
+    else:
+        segment_outcomes = [
+            {
+                "segment_index": idx,
+                "summary": segment.description,
+                "success": segment.success,
+            }
+            for idx, segment in enumerate(segments)
+        ]
+
+    # Use LLM-provided session outcome
+    session_outcome = {
+        "description": analysis.session_outcome.description,
+        "success": analysis.session_outcome.success,
+    }
+
+    return {
+        "segments": summary_segments,
+        "key_actions": key_actions,
+        "segment_outcomes": segment_outcomes,
+        "session_outcome": session_outcome,
+    }
+
+
+def _convert_video_segments_to_session_summary_fallback(
```

Not sure about this part. If something failed this miserably - we probably need just to fail completely, raise an exception, and then update the solution to not fail, instead of trying to return something minimally usable. Up to the discussion though.